frontend/packages/rhoas-plugin/src/catalog/providers/useRhoasCatalog.tsx

```diff
@@ -0,0 +1,143 @@
+
+    const tokenStatusFooter = () => {
+      let token;
+      if (serviceAccount === null || !isServiceAccountValid) {
+        token = (
+          <Label variant="outline" color="orange" icon={<LockIcon />}>
+            {t('rhoas-plugin~Unlock with token')}
+          </Label>
+        );
+      } else {
+        token = t('rhoas-plugin~Unlocked');
+      }
+      return (
+        <Flex direction={{ default: 'column' }}>
+          <FlexItem>
+            {t('rhoas-plugin~RHOAS can include Streams for Kafka, Service Registry')}
+          </FlexItem>
+          <FlexItem>{token}</FlexItem>
+        </Flex>
+      );
+    };
+
+    const serviceKafkaCardDescription = (
+      <TextContent>
+        <Text component={TextVariants.p}>{t('rhoas-plugin~KafkaCardDescription')}</Text>
+      </TextContent>
+    );
+
+    const cloudServicesCardDescription = (
+      <Flex direction={{ default: 'column' }}>
+        <FlexItem>
+          <TextContent>
+            <Text component={TextVariants.p}>
+              {t('rhoas-plugin~ManagedServices-card-description')}
+            </Text>
+          </TextContent>
+        </FlexItem>
+        <Divider component="li" />
+        <FlexItem>
+          <Text component={TextVariants.h2}>
+            {t('rhoas-plugin~Access Red Hat Cloud Services with API Token')}
+          </Text>
+          <ServiceToken namespace={namespace} />
+        </FlexItem>
+      </Flex>
+    );
+
+    const serviceKafkaCardDetailsDescription = [
+      {
+        value: serviceKafkaCardDescription,
+      },
+    ];
+
+    const cloudServicesCardDetailsDescription = [
+      {
+        value: cloudServicesCardDescription,
+      },
+    ];
+
+    if (isServiceAccountValid) {
+      const serviceKafkaCard: CatalogItem[] = [
+        {
+          name: 'Streams for Apache Kafka',
+          type: CATALOG_TYPE,
+          uid: 'streams-1615213269575',
+          description: 'Streams for Apache Kafka',
+          provider: 'Red Hat',
+          tags: ['kafka', 'service'],
+          icon: {
+            url: kafkaIcon,
+          },
+          cta: {
+            label: 'Connect',
+            href: `/rhoas/ns/${namespace}/kafka`,
+          },
+          details: {
+            descriptions: serviceKafkaCardDetailsDescription,
+          },
+        },
+      ];
+      return serviceKafkaCard;
+    }
+
+    const cloudServicesCard: CatalogItem[] = [
+      {
+        name: 'Red Hat Cloud Services',
+        type: CATALOG_TYPE,
+        uid: 'services-1615213269575',
+        description: tokenStatusFooter(),
+        provider: 'Red Hat',
+        tags: ['kafka', 'service'],
+        icon: {
+          url: operatorIcon,
+        },
+        details: {
+          descriptions: cloudServicesCardDetailsDescription,
+        },
+      },
+    ];
+    return cloudServicesCard;
+  }, [namespace, errorMsg, isServiceAccountValid, loaded, serviceAccount, t]);
+
+  return [services, loadedOrError, undefined];
```

Comment:
This was initially there but because we have this transformation tile when CR is not there it will show constant loading screen. I spend some time trying to figure out more elegant solution, but this seems like it is the best I could come with

Reply:
@wtrocki you're right. If there's an error, you actually show a different card. In which case it isn't an error we need to present to the catalog.

Ignore my suggestion!